Analysis/src/EventTags.cpp

```diff
@@ -78,15 +84,22 @@ EventTags EventTagCreator::CreateEventTags(const DataId& dataId_base, float weig
         { DiscriminatorWP::Medium, weight_btag_Medium },
         { DiscriminatorWP::Tight, weight_btag_Tight },
     };
-    const auto get_weight_btag = [&](DiscriminatorWP wp, UncertaintySource unc_source, UncertaintyScale unc_scale) {
+
+    const auto get_weight_btag = [&](DiscriminatorWP wp, UncertaintySource unc_source, UncertaintyScale unc_scale, bool tune) {
         if(use_IterativeFit) {
-            if(iterativeFit_corrections.count(std::make_pair(unc_source, unc_scale))){
-                return weight_btag_IterativeFit * iterativeFit_corrections.at({unc_source, unc_scale});
+            btag_corrections parameters;
+            parameters.unc_source = unc_source;
+            parameters.unc_scale = unc_scale;
+            parameters.tune = tune;
+            if(!iterativeFit_corrections.count(parameters)){
+                parameters.unc_source = UncertaintySource::None;
+                parameters.unc_scale = UncertaintyScale::Central;
+                parameters.tune = 0;
```

Comment:
First, try without resetting the `tune`. If still not found reset tune to `false`.